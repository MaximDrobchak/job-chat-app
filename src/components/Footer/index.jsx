import React, { Component } from 'react';
import injectSheet from 'react-jss';
import PropTypes from 'prop-types';

const styles = {
	root: {
		padding: {
			top: 20,
			bottom: 20,
			left: 20,
			right: 20,
		},
		'& div': {
			marginTop: 20,
			display: 'flex',
			justifyContent: 'space-between',
			'& span': {
				opacity: 0.5,
				fontFamily: 'Roboto',
				fontSize: 14,
			},
			'& blockquote': {
				marginTop: 0,
				opacity: 0.5,
				fontFamily: 'Roboto',
				fontSize: 14,
			},
		},
	},
};
class Footer extends Component {
	render() {
		const { classes, children } = this.props;
		return (
			<div className={classes.root}>
				{children}
				<div>
					<span>
						Enter to send a message <strong>Shift + Enter</strong> for new line
					</span>
					<blockquote>
						<b>*bold </b>
						<i>_italic_ </i>
						<strike>~strice~ </strike>
						>qoute
					</blockquote>
				</div>
			</div>
		);
	}
}

Footer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.object.isRequired,
};
export default injectSheet(styles)(Footer);
