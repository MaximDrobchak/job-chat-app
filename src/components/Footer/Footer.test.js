import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Footer from './index.jsx';

describe('Footer', () => {
	it('отрисовывает без ошибки', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Footer />, div);
		ReactDOM.unmountComponentAtNode(div);
	});
	test('есть корректный снимок', () => {
		const component = renderer.create(<Footer />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
