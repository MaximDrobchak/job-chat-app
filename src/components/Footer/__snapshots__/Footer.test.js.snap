// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer есть корректный снимок 1`] = `
<div
  className="Footer-root-0-1-1"
>
  <div
    className="MessageInput-root-0-1-2"
  >
    <input
      placeholder="Message to Terry Crews"
      type="text"
    />
    <span
      className="MessageInput-buttonMessage-0-1-3"
      onClick={[Function]}
    >
      <img
        alt="attachment"
        src="attachment.svg"
      />
    </span>
  </div>
  <div>
    <span>
      Enter to send a message 
      <strong>
        Shift + Enter
      </strong>
       for new line
    </span>
    <blockquote>
      <b>
        *bold 
      </b>
      <i>
        _italic_ 
      </i>
      <strike>
        ~strice~ 
      </strike>
      &gt;qoute
    </blockquote>
  </div>
</div>
`;
